packages/sdk/src/accountTransactions.ts

```diff
@@ -845,51 +845,50 @@ export class UpdateCredentialKeysHandler
+        const serializedCredId = CredentialRegistrationId.toBuffer(payload.credId);
 
         const keys = payload.keys.keys;
 
         const putCredentialVerifyKey = (credentialVerifyKey: Upward<VerifyKey>) => {
-            if (isKnown(credentialVerifyKey)) {
-                const serializedSchemeId = encodeWord8FromString(credentialVerifyKey.schemeId);
-                const serializedKey = Buffer.from(credentialVerifyKey.verifyKey);
-                return Buffer.concat([serializedSchemeId, serializedKey]);
-            }
-            return Buffer.alloc(0); //TODO: we are dealing with Upward, would this be ok?
+            if (!isKnown(credentialVerifyKey)) throw new Error('Verify keys must be known when serializing');
+
+            const serializedSchemeId = encodeWord8FromString(credentialVerifyKey.schemeId);
+            const serializedKey = Buffer.from(credentialVerifyKey.verifyKey, 'hex');
+            return Buffer.concat([serializedSchemeId, serializedKey]);
         };
+        console.log('----> about to serializeMap');
+        console.log('===', keys);
+        const serializedKeys = serializeMap(keys, encodeWord8, encodeWord8FromString, putCredentialVerifyKey);
 
-        const serializedKeys = serializeMap(keys, encodeWord8, encodeWord8FromString, putCredentialVerifyKey); //TODO: the map key is of type number, would the encodeWord8FromString work?
+        const serializedThreshold = Buffer.from([payload.keys.threshold]);
 
-        return Buffer.concat([serializedCredId, serializedKeys]);
+        return Buffer.concat([serializedCredId, serializedKeys, serializedThreshold]);
     }
 
     deserialize(serializedPayload: Cursor): UpdateCredentialKeysPayload {
-        //TODO: is this correct?
         const credId = serializedPayload.read(48).toString('hex');
-
+        console.log('----> credId: ', credId);
         const resultMap: Record<number, Upward<VerifyKey>> = {};
         const credVerifyKeyEntryCount = serializedPayload.read(1).readUInt8(0);
+        console.log('----> count: ', credVerifyKeyEntryCount);
         for (let a = 0; a < credVerifyKeyEntryCount; a++) {
             const index = serializedPayload.read(1).readUInt8(0);
-
+            console.log('----> index:', index);
             const scheme = serializedPayload.read(1).readUInt8(0);
-            const key = serializedPayload.read(32);
-
-            const currentVerifyKey = {
-                schemeId: scheme.toString(),
-                verifyKey: key.toString('hex'), //TODO: is this correct, i do toString hex?
-            };
-
+            console.log('----> schemeId: ', scheme);
+            let currentVerifyKey: Upward<VerifyKey> = null;
+            if (scheme === 0) {
+                currentVerifyKey = {
+                    schemeId: '0', //TODO: are we passing in 0 or Ed25519 during serialization or do we convert Ed25519 to 0 and vice versa?
+                    verifyKey: serializedPayload.read(32).toString('hex'),
+                };
```

Comment:
```suggestion
                currentVerifyKey = {
                    schemeId: 'Ed25519',
                    verifyKey: serializedPayload.read(32).toString('hex'),
                };
```

Needs to be 'Ed25519' AFAIK.